mikeio1d/result_reader_writer/result_reader.py

```diff
@@ -80,7 +82,7 @@ class ResultReader(ABC):
     def __init__(
         self,
         res1d,
-        file_path=None,
+        file_path: str | Path = None,
```

Comment:
@ryan-kipawa It could be the same problem when calling Res1D.save, because initially Path objects were not considered as an input. I suggest to check everywhere where there is Connection referenced to make similar update.